src/components/RewardWrapper/index.tsx

```diff
@@ -214,15 +221,27 @@ export function RewardWrapper({
 
     let address: `0x${string}` | undefined
 
-    if (account.address) {
+    /**
+     * handles https://github.com/HyperPlay-Gaming/product-management/issues/801
+     * Sometimes wagmi does not establish a connection but useAccount returns the address.
+     * We need to check that the switch chain method exists before proceeding with claiming.
+     */
+    let connectionHasSwitchChain = false
+    if (config.state.current) {
+      const currentConnection = config.state.connections.get(
+        config.state.current
+      )
+      connectionHasSwitchChain = !!currentConnection?.connector.switchChain
+    }
+
+    if (account.address && connectionHasSwitchChain) {
       address = account.address
     } else {
       onShowMetaMaskPopup?.()
```

Comment:
I don't think this is needed, but didn't want to slow down this PR. If MM ext sends a notification, the appropriate window will already be shown in the UI.